packages/forms/src/components/DateInput/ControlledInput.tsx

```diff
@@ -39,7 +38,7 @@ const ControlledInput = ({
   const intl = useIntl();
   const inputStyles = useInputStyles();
   const selectStyles = useInputStyles("select");
-  const rawDefaultValue: unknown = get(defaultValues, name);
+  const rawDefaultValue: unknown = defaultValues?.[name];
```

Comment:
Oh shoot sorry, I think this is causing a bug with field arrays since it is nested.

```ts
// Looking for this
const defaultValues = {
  'date.0': '2020-01-02'
};

// Actually this
const defaultValues = {
  data: [ '2020-01-02' ]
};
```

If we want complex paths, you could use this: https://youmightnotneed.com/lodash#get 🤔

Reply:
Please don't add any new helper functions in this PR.  If you can't accomplish it with built-in language features then it should be out of scope.

Reply:
Ok, I'm not sure how to do this without adding a helper function so I've [reverted](https://github.com/GCTC-NTGC/gc-digital-talent/pull/13398/commits/85b193c4ab5ea1a3e4d3180fdd40c312864ee1dd) to the lodash function here instead.